integration/rest/src/main/java/it/gov/pagopa/rtd/transaction_filter/logger/MaskingPatternLayout.java

```diff
@@ -0,0 +1,48 @@
+package it.gov.pagopa.rtd.transaction_filter.logger;
+
+import ch.qos.logback.classic.PatternLayout;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+
+public class MaskingPatternLayout extends PatternLayout {
+
+  private Pattern multilinePattern;
+  private final List<String> maskPatterns = new ArrayList<>();
+
+  // invoked for every single entry in the xml
+  public void addMaskPattern(String maskPattern) {
+    maskPatterns.add(maskPattern);
+    multilinePattern = Pattern.compile(String.join("|", maskPatterns),
+        Pattern.MULTILINE);
+  }
+
+  @Override
+  public String doLayout(ILoggingEvent event) {
+    return maskMessage(super.doLayout(event));
+  }
+
+  private String maskMessage(String message) {
+    if (multilinePattern == null) {
```

Comment:
@GiovanniMancini I couldn't manage to cover this condition in the unit tests. It needed to load a logback configuration file without `<maskPattern>` nodes. Spring boot loads `logback-test.xml` as default and i failed to load a different file through `logging.config` (with `@TestPropertySource`). 
I tried to use `<springProfile>` to differentiate within a single config file but it didn't work too.